core/src/main/java/org/apache/calcite/rel/rules/AggregateCaseToFilterRule.java

```diff
@@ -227,8 +227,11 @@ protected AggregateCaseToFilterRule(RelBuilderFactory relBuilderFactory,
           RelCollations.EMPTY, aggregateCall.getType(),
           aggregateCall.getName());
     } else if (kind == SqlKind.SUM // Case B
-        && isIntLiteral(arg1) && RexLiteral.intValue(arg1) == 1
-        && isIntLiteral(arg2) && RexLiteral.intValue(arg2) == 0) {
+        && isIntLiteral(arg1)
+        && RexLiteral.intValue(arg1) == 1
```

Comment:
Do we need to check arg1 against null literal to avoid NPE for the following case?

sum(case when deptno = 40 then null end) as count_d40

Reply:
good point, I think we should

Reply:
> Do we need to check arg1 against null literal to avoid NPE for the following case?
> 
> sum(case when deptno = 40 then null end) as count_d40

There's no NPE, UT is added.